trusted-committer-article.md

```diff
@@ -0,0 +1,218 @@
+    - It is important to both give and take in the community
+    - If there are people toxic to the communities health, it is the TCs
+      responsibility to either try and change, contain or remove those people
+    - Communicate the purpose of the community and try to make everyone on
+      board with the purpose.
+    - Laude excellent or first contributions
+    - Make sure community members have a chance to level up by suggesting 
+      challenging tasks.
+    - Commmunicate technical and community risks to responsible managers
+    - Help potential community members with them beeing empowered to make
+      contributions by talking to their managers
+    - Ensure there are users and potential contributors by promoting and
+      advertising the community and its goals
+    - Make sure people get a together regularly
+      
+## Reducing Barriers To Entry
+
+- Soliciting contributions is one of the things that is more challenging with
+  InnerSource compared to Open Source. There are two main reasons for that 
+    - The sheer number of potential contributors is lower in InnerSource
+    - Contributors will want to contribute during their work time. That means
+      they are more time constrained compared to doing Open Source after office
+      hours.
+- It is therefore super important to make the process for making contributions
+  and for onboarding contributors as frictionless as possible to avoid wasting
+  a contributors time. 
+- There are a number of things a TC can do to achieve that state
+    - have a good readme.md, providing detailed instructions on how to get,
+      build, test and use the SW developed by the InnerSource community.
+    - have a good contribute.md which outlines what is expected of the
+      contributor to make a contribution. It should answer common questions
+      like
+        - Who do I contact in case of questions and how to reach her?
+        - How do I submit a bug report or feature request?
+        - What is the definition of done for a contribution; code, tests, docs?
+        - What are conventions for code style, branching or committ messages, e. g.?
+        - What is expected of me in terms of supporting contributed code after
+          the contribution was accepted?
+        - What are the process steps that govern contributions?
+        - Is there a code of conduct or other guidelines to how the community
+          operates?
+- As a TC, I have to make sure there is someone in the community who is
+  available to answer questions of users and potential contributors.
+- Actively publishing a list of "Newbie Tasks" to actively solicit contributions,
+  rather than waiting for them to happen.
+- Also make it clear that non-code contributions are welcome, such as writing
+  documentation, organizing events or creating artwork.
+- Gig-Marketplace as a good example of how to do this.
+
+## Uplevelling Community Members
```

Comment:
I wrote up something on this at one point... let me try to dig it up.

Reply:
This was targeted at communities of practice:
But I think the same ideas broadly apply and TC's fall into the maintainers and driver buckets of this continuum.

## The people
Individuals joining your community can be described by their expected outcome of participation.  This forms a similar continuum:

* Unaware
* Newcomer
* Consumer
* Contributor
* Maintainer
* Driver

#### Unaware

These individuals haven't heard about your effort or don't share the same problems as your community.  

#### Newcomer

This is someone who might be a first-time participant.  They may have just become aware of your project but haven't spent much time learning about the context, motivations and history of the project.

#### Consumer 

This person is generally interested in deriving value passively from exisiting projects.  They're on the lookout to learn about how things work, and if a project solves a particular problem they're faced with.  As a result they're interested in best practices and breaking changes.  They don't have time to be deeply involved.  What will motivate this person to become an active participant? Does the overall plan for the community need adjusting?  Perhaps some documentation, a workshop or project scaffolding can help them become a contributor.

#### Contributor

This is a person that contributes code or often takes the driver's seat in the community.  They have opinions and are passionate about how the work is completed.

#### Maintainer

This person is committed to the effort and is willing to take on some of the responsibility
of keeping a community effort afloat. They might help find other presenters or set up content
for discussion.  This person can also aid the facilitator by keeping a pulse on the activity
of community members. 

#### Driver

This person sets the greater vision, and works to develop a plan to attain the vision.  
They setup meetings that help move the mission forward, meetings that fulfill the 
vision of the community.  In addition, this person keeps a pulse on new projects to
see how they may or may not roll into the bigger picture of the community.  

## Strategy

Regular and systematic contact with individuals helps move them along the continuum of participation. 
Often it takes ***real work*** to identify and understand what it takes to help someone level up.  

Generally, the first members are innovators and early adopters. These people are passionate about working
together to spread new ideas and improve the world.  Engage them to get help in prototyping and putting forward something tangible to bring to the early majority.

The early majority will start off as consumers, their interests are to gather information and
stay current.  It is important to keep your early adopters engaged so that they can help you brige the gap to the early majority.  Eventually, you will end up with a group of people who consistently and regularly meet together that fall within the general majority (both the early and late majority).  This becomes the core of your community.

### Leveling-up the Community

Unaware

* Think of alternative ways to market your community.  Using a personal touch can make a big difference.

Newcomer

* Orient the individual to the community effort.  A newcomer will not have the same context as the existing members.
* Share relevant success stories, this will build confidence that your community is making progress.
* Ask questions to understand the newcomers own goals and interests.
* Think about and determine what future opportunities you might pursue together.

Consumer

* Determine ahead of time the roles that are available, and let the participant know what needs doing. 
* Understand and ask them what it would take for them to contribute or participate in an active way.
* Ask questions to understand the participant own goals and interests.
* If there are a few consumers who are close to participating, consider updating the general plan.  Adjust the backlog of work for the community to incorporate work that aligns with the individuals own goals an interests.

Contributor

* Understand and ask them what it would take for them to contribute to the mission or vision of the group.
* Be aware of the persons "day" job.  Understand how it relates to the community plan and effort.  Think about ways to get work done that contributes to their "day" job, as opposed to their 10% time.

Maintainer

* Work to define ahead of time the roles that are available. 
* Create or find small roles that this person can fulfill to aide the facilitator.
* Find roles and tasks that aid in growing the community.

## Hitting your stride

The ideal state for a community is to have a regular amount of consumers and contributors involved, as well as a regular number of maintainers.  This group doesn't have to be large, but it does need to have a good range of individuals spanning this continuum to be healthy.  

> Never doubt that a small group of thoughtful, committed citizens can change the world; 
> indeed, it's the only thing that ever has. - Margaret Mead

Reply:
> […] move them along the continuum of participation.

I like that!

Also like that you make it about upleveling _the community_ rather than about upleveling _individual contributors_.